fem/fe_coll.hpp

```diff
@@ -153,14 +153,54 @@ class FiniteElementCollection
       SDim <= Dim(Geom). */
    void SubDofOrder(Geometry::Type Geom, int SDim, int Info,
                     Array<int> &dofs) const;
+
+   /// Variable order version of FiniteElementForGeometry
+   const FiniteElement *GetFE(Geometry::Type geom, int p) const
+   {
+      if (p == base_p) { return FiniteElementForGeometry(geom); }
+      if (p >= var_orders.Size() || !var_orders[p]) { InitVarOrder(p); }
+      return var_orders[p]->FiniteElementForGeometry(geom);
+   }
+
+   /// Variable order version of DofForGeometry
+   int GetNumDof(Geometry::Type geom, int p) const
+   {
+      if (p == base_p) { return DofForGeometry(geom); }
+      if (p >= var_orders.Size() || !var_orders[p]) { InitVarOrder(p); }
+      return var_orders[p]->DofForGeometry(geom);
+   }
+
+   /// Variable order version of DofOrderForOrientation
+   const int * GetDofOrdering(Geometry::Type geom, int p, int ori) const
+   {
+      if (p == base_p) { return DofOrderForOrientation(geom, ori); }
+      if (p >= var_orders.Size() || !var_orders[p]) { InitVarOrder(p); }
+      return var_orders[p]->DofOrderForOrientation(geom, ori);
+   }
+
+   /// Return the order the FE collection was constructed with.
+   int GetOrder() const { return base_p; }
+
+protected:
+   const int base_p;
+
+   FiniteElementCollection() : base_p(0) {}
+   FiniteElementCollection(int p) : base_p(p) {}
+
+   /// Instantiate a new collection of the same type with a different order.
+   virtual FiniteElementCollection* CloneOrder(int p) const;
```

Comment:
Maybe `CloneWithOrder` would be clearer?

Reply:
I'd prefer the simpler name...

Reply:
I also prefer the more self documented name `CloneWithOrder`

Reply:
I didn't expect this non-public method to be controversial. I guess both names are acceptable, but at the same time both are not optimal: in the short version, "order" is supposed to mean the whole sub-collection (this might be confusing) while in the longer version "order" refers to the integer parameter, which I find a bit clumsy. The short version has the benefit of at least being short, which is why I prefer it.

Reply:
`Clone` is fine with me.